rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,179 @@
+↖️ Table of Contents
+
+# RFC: Monorepo
+
+**Status:** 🚧 WIP, comments are welcome nonetheless
+
+## Reviewers
+
+- [ ] @zackkrida
+- [ ] @sarayourfriend
+
+## Rationale
+
+For a comprehensive discussion about the pros, the cons and the counterpoints to each see [discussion](https://github.com/WordPress/openverse/issues/192). This is not the purpose of this RFC.
+
+This RFC summarily lists the benefits and then, with the twin assumptions of a monorepo being ultimately beneficial and the decision to migrate being finalised in the above discussion, proceeds to go into the implementation details.
+
+### Exclusive benefits of monorepo
+
+This only includes things that cannot be accomplished without the use of a monorepo.
+
+1. Single place to go for issues, PRs and all activity. Currently tickets are scattered across several repos, and any tickets that could benefit more than a single layer must be opened in each of the different repos.
+
+1. Singular copy (different from synced independent copies) of scaffolding code such as Git hooks, lint rules and common workflows. This is distinctly better than elaborate sync workflows.
+
+1. Central place for all technical documentation, enabling documentation for different parts of the stack to cross-reference other pieces and stay current with changes in other places.
```

Comment:
What is wrong with old normal links? We can already cross-reference info if it's necessary.

Reply:
Hyperlinks can break when documentation is reorganised or reworded. Having references to the changed docs being in the same repo makes it easy for the PR author to update it as a part of the same PR and prevent the links from rotting or breaking. 

[Sphinx explains it better than I can](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#cross-referencing-with-sphinx).

Reply:
This has been a pain point for documenting our data models and ingestion/data refresh processes. IMO that's one concrete place where documentation (and the process of generating/updating that documentation) could improve in a monorepo.